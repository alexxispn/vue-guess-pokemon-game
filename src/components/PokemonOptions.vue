<template>
    <ul class="options">
      <li v-for="pokemon in pokemons"
          :key="pokemon.id"
          @click="$emit('selectedPokemon', pokemon.id)">
        {{ pokemon.name }}
      </li>
    </ul>
</template>

<script>
export default {
  name: "PokemonOptions",
  props: {
    pokemons: {
      type: Array,
      required: true
    }
  }
}
</script>

<style scoped>
.options {
  display: flex;
  flex-direction: column;
  place-items: center;
  list-style: none;
  gap: clamp(0.8rem, 2vw, 1.6rem);
  width: 80%;
}
</style>
