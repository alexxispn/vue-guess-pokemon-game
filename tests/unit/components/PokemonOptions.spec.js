import PokemonOptions from "@/components/PokemonOptions";
import {shallowMount} from "@vue/test-utils";
import {PokemonsMock} from "../mocks/pokemons.mock";

describe('PokemonOptions', () => {
    const wrapper = shallowMount(PokemonOptions, {
        props: {
            pokemons: PokemonsMock
        }
    })
    const li = wrapper.findAll('li')
    it('should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('should return 4 li with the correct pokemon names', () => {
        expect(li.length).toBe(4)
        expect(li[0].text()).toBe('Bulbasaur')
        expect(li[1].text()).toBe('Ivysaur')
        expect(li[2].text()).toBe('Venusaur')
        expect(li[3].text()).toBe('Charmander')
    })
    it('should emit "selectedPokemon" with the pokemon.id when click on li',() => {
        li[0].trigger('click')
        li[1].trigger('click')
        li[2].trigger('click')
        li[3].trigger('click')
        expect(wrapper.emitted('selectedPokemon').length).toBe(4)
        expect(wrapper.emitted('selectedPokemon')[0]).toEqual([1])
        expect(wrapper.emitted('selectedPokemon')[1]).toEqual([2])
        expect(wrapper.emitted('selectedPokemon')[2]).toEqual([3])
        expect(wrapper.emitted('selectedPokemon')[3]).toEqual([4])
    })
    it('',() => {

    })
})
