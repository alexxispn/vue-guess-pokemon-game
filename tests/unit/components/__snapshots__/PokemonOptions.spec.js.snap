// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions should match with snapshot 1`] = `
<ul class="options">
  <li>Bulbasaur</li>
  <li>Ivysaur</li>
  <li>Venusaur</li>
  <li>Charmander</li>
</ul>
`;
